src/dialogs/quick-open/ha-quick-open-dialog.ts

```diff
@@ -0,0 +1,220 @@
+
+    return reloadableDomains.map((domain) => ({
+      text:
+        this.hass.localize(`ui.dialogs.quick_open.commands.${domain}`) ||
+        this.hass.localize(
+          "ui.dialogs.quick_open.commands.reload",
+          "domain",
+          domainToName(this.hass.localize, domain)
+        ),
+      domain,
+      onClick: this._executeCommand,
+    }));
+  }
+
+  private _generateEntityItems(
+    hass: HomeAssistant,
+    _filter: string
+  ): QuickOpenItem[] {
+    return Object.keys(hass.states)
+      .map((key) => hass.states[key])
+      .map<QuickOpenItem>((entity) => ({
+        text: entity.entity_id,
+        domain: computeDomain(entity.entity_id),
+        onClick: this._entityMoreInfo,
+      }));
+  }
+
+  private _filterItems(
+    items: QuickOpenItem[],
+    filter: string
+  ): QuickOpenItem[] {
+    return items
+      .filter((value) => fuzzySequentialMatch(filter.toLowerCase(), value.text))
+      .sort((itemA, itemB) => {
+        if (itemA.text < itemB.text) {
+          return -1;
+        }
+        if (itemA.text > itemB.text) {
+          return 1;
+        }
+        return 0;
+      });
+  }
+
+  private async _executeCommand(ev: Event) {
+    const eventData = {
+      domain: (ev.target as any).domain,
+      service: "reload",
+    };
+
+    await this.hass.callService(eventData.domain, "reload");
+
+    fireEvent(this, "hass-service-called", eventData);
+
+    this.closeDialog();
+  }
+
+  private _entityMoreInfo(ev: Event) {
+    ev.preventDefault();
+    fireEvent(this, "hass-more-info", {
+      entityId: (ev.target as any).text,
+    });
+    this.closeDialog();
+  }
+
+  protected render() {
+    if (!this._opened) {
+      return html``;
+    }
+
+    return html`
+      <ha-dialog open @closed=${this.closeDialog} hideActions>
+        <paper-input
+          dialogInitialFocus
+          no-label-float
+          @value-changed=${this._entityFilterChanged}
+          .label=${this.hass.localize(
+            "ui.dialogs.quick_open.filter_placeholder"
+          )}
+          type="search"
+          value=${this._commandMode
+            ? `>${this._entityFilter}`
+            : this._entityFilter}
+        ></paper-input>
+        <mwc-list>
+          ${this._items.map(
+            ({ text, domain, onClick }) => html`
+              <mwc-list-item
+                @click=${onClick}
+                .domain=${domain}
+                alt=${this.hass.localize(
+                  "ui.panel.developer-tools.tabs.states.more_info"
+                )}
+                .title=${this.hass.localize(
+                  "ui.panel.developer-tools.tabs.states.more_info"
+                )}
+              >
+                <ha-icon .icon=${domainIcon(domain)}></ha-icon>
+                ${text}
+              </mwc-list-item>
+            `
+          )}
+        </mwc-list>
+      </ha-dialog>
+    `;
+  }
+
+  static get styles() {
+    return [
+      haStyleDialog,
+      css`
```

Comment:
We should add some on hover here. When hovering with the mouse over a list item change the background color. Unless it does that already. Maybe mimick the ha-table hover?

Reply:
Yeah, the background already changes on hover. Looks like the GIF capture tool compressed that out of existence. But it's there.

![image](https://user-images.githubusercontent.com/1480827/95137133-4e5b0a00-071c-11eb-895e-e8e9e629d517.png)

(Don't mind the lack of icons... I'm currently trying to get graphic items to work)